### PR TITLE
Update network loader to support mainnet

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

This pull request updates the network loader to support mainnet. This pull request also updates the `getNetworkClient` method to support a `mainnet` option.

**Changes** 🏗

* Added `mainnet` contract address to both versions of the `EtherRouter` contract.
* Updated the network loader to load contracts by version only. We no longer need to load contracts by network because `goerli` and `mainnet` are identical and `rinkeby` is no longer supported.
* Updated the network loader tests to include tests for both `goerli` and `mainnet`.

### CODE REVIEW

File: packages/colony-js-client/src/getNetworkClient.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const defaultInfuraProjectId = '7d0d81d0919f4f05b9ab6634be01ee73';
 const getInfuraProvider = (network: string, infuraProjectId?: string) => {
   let host;
   switch (network) {
-    case 'homestead':
+    case 'mainnet':
```
Comment (sprusr) [inline packages/colony-js-client/src/getNetworkClient.js]: Is `homestead` what others use? I think it's best to go with the crowd here. Maybe we should allow either, and also possibly default to this?

File: packages/colony-js-contract-loader-network/contracts/versioned/v1/EtherRouter.json [hunk 2]
```diff
@@ -132,6 +132,9 @@
   "networks": {
     "goerli": {
       "address": "0x79073fc2117dD054FCEdaCad1E7018C9CbE3ec0B"
+    },
+    "mainnet": {
```
Comment (sprusr) [inline packages/colony-js-contract-loader-network/contracts/versioned/v1/EtherRouter.json]: Same with this, `homestead` or `mainnet` - I don't know which is technically correct, or even if you could say one is more correct than the other.

File: packages/colony-js-client/src/getNetworkClient.js (hunk 1 above)
Comment (ryanchristo) [reply packages/colony-js-client/src/getNetworkClient.js]: Yea, probably safe to allow both. I was not entirely sure myself.

File: packages/colony-js-contract-loader-network/contracts/versioned/v1/EtherRouter.json (hunk 2 above)
Comment (ryanchristo) [reply packages/colony-js-contract-loader-network/contracts/versioned/v1/EtherRouter.json]: I will add both just to be safe. 👍

Comment (rdig) [reply packages/colony-js-contract-loader-network/contracts/versioned/v1/EtherRouter.json]: I use `homestead` in `purser`: https://github.com/JoinColony/purser/blob/master/modules/node_modules/%40colony/purser-core/defaults.js#L126

But in the `purser` context it's a bit irrelevant as I only address networks by `id` and not by name

**Pull request merged**